core/spi.py

```diff
@@ -220,17 +229,18 @@ def read_spi_readoutmode(self):
         """ Continous readout """
         pass
 
-    def read_spi_fifo(self) -> bytearray:
-        """ Read Data from SPI FIFO until empty """
-
+    def read_spi_fifo(self, max_reads: int = 1) -> bytearray:
+        """ Read Data from SPI FIFO until empty
+        :param max_reads: Max read cycles
+        :returns: SPI read stream
+        """
         read_stream = bytearray()
+        readcount = 0
 
-        while not self.get_spi_config() & 16:
-            readbuffer = self.read_spi(64)
-
+        while not (self.get_spi_config() & SPI_READ_FIFO_EMPTY) and readcount<max_reads:
+            readbuffer = self.read_spi(1024)
```

Comment:
I did some further testing, increasing this to 4096 resulted in a significant performance improvement.
With a simple testscript with just one loop for readout, decoding and writing to a file, I can process 70k Hits per second.

while time.time() < stop_time:
        readout = nexys.read_spi_fifo(10)

        if readout != b'':
            cnt += 1
            print(cnt)

            list_hits = decode.hits_from_readoutstream(readout)
            cnt_list_hits += len(list_hits)
            print(cnt_list_hits)
            df = decode.decode_astropix2_hits(list_hits)
  
            df.to_csv(output_path, mode='a', header=not os.path.exists(output_path))